### ISSUE TITLE
Upgrade python to 3.11.1

### ISSUE THREAD

**Issue by mkoeppe** [sagemath/sage#34830]
<div id="comment:0"></div>

https://docs.python.org/3/whatsnew/changelog.html#changelog

Depends on #33842

Component: **packages: standard**

_Issue created by migration from https://trac.sagemath.org/ticket/34830_

**Comment by mkoeppe** [comment]
Branch: **[u/mkoeppe/upgrade_python_to_3_11_1](https://github.com/sagemath/sagetrac-mirror/tree/u/mkoeppe/upgrade_python_to_3_11_1)**

**Comment by mkoeppe** [comment]
Author: **Matthias Koeppe**

**Comment by mkoeppe** [comment]
<div id="comment:2"></div>

Last 10 new commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/fd4d697579053a65c7504ee0798fcb9c9e88a9a2"><code>fd4d697</code></a></td><td><code>src/sage/misc/fpickle.pyx: Fix docstring markup</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/d7b029436482fd3ef264892effc8d92dbfdad539"><code>d7b0294</code></a></td><td><code>build/pkgs/python3/spkg-configure.m4: Accept system python 3.11</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/25c9ad66ae8ac790b8c41e5dc74ef1f72b06a912"><code>25c9ad6</code></a></td><td><code>Merge tag '9.8.beta4' into t/33842/py311</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/7099e5147174bce8dabe8b2a8ef5931ac60e7f50"><code>7099e51</code></a></td><td><code>Revert "build/pkgs/cypari/patches/trashcan.patch: Remove"</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/8df4d8b64c62822235748a2b19e64c1d7a8fee31"><code>8df4d8b</code></a></td><td><code>Revert "build/pkgs/cython/patches/trashcan.patch: Remove"</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/0e458dbd362924aa8f73c6613b566b94e1d42b24"><code>0e458db</code></a></td><td><code>build/pkgs/cython/patches/trashcan.patch: Update using https://github.com/cython/cython/pull/4475</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/bc69b339108b007615f9768f273bf03af273c316"><code>bc69b33</code></a></td><td><code>build/pkgs/babel: Update to 2.11.0</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/c928c2c79788631da240ae0c3713aeafad43854d"><code>c928c2c</code></a></td><td><code>src/sage/all.py: Filter out imghdr deprecation warnings</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/a79c6b58a66d4f8e9976958dc7e39987e07145fa"><code>a79c6b5</code></a></td><td><code>Merge #33842</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/aab67fa6215d174a567f580122dff75294e56a84"><code>aab67fa</code></a></td><td><code>build/pkgs/python3: Update to 3.11.1</code></td></tr></table>

**Comment by mkoeppe** [comment]
Commit: **[`aab67fa`](https://github.com/sagemath/sagetrac-mirror/commit/aab67fa6215d174a567f580122dff75294e56a84)**

**Comment by mkoeppe** [comment]
Changed commit from **[`aab67fa`](https://github.com/sagemath/sagetrac-mirror/commit/aab67fa6215d174a567f580122dff75294e56a84)** to none

**Comment by mkoeppe** [comment]
<div id="comment:3" align="right">comment:3</div>

merged into #33842

**Comment by mkoeppe** [comment]
Changed author from **Matthias Koeppe** to none

**Comment by mkoeppe** [comment]
Changed branch from **[u/mkoeppe/upgrade_python_to_3_11_1](https://github.com/sagemath/sagetrac-mirror/tree/u/mkoeppe/upgrade_python_to_3_11_1)** to none
